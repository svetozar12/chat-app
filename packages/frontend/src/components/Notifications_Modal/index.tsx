export { default } from "./Notifications_Modal";
