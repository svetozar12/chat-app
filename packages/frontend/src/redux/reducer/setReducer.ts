import { Action, ActionType, InitialState2 } from "../state";
export const initialState = {
  reciever: "",
  chat_inviter: "",
  pageNumber: 2,
  setUserSettings: false,
  setFriendRequest: false,
  toggleCreateGroup: false,
  setMobileNav: false,
  setIsMatch: false,
  setChatSettings: false,
  setModalInvite: false,
  setIsLoggingIn: false,
};
const setReducer = (state: InitialState2 | any | InitialState2[] = initialState, action: Action) => {
  switch (action.type) {
    case ActionType.SET_RECIEVER:
      return { ...state, reciever: action.payload };
    case ActionType.CHAT_INVITER:
      return { ...state, chat_inviter: action.payload };
    case ActionType.INCREMENT_PAGE_NUMBER:
      return { ...state, pageNumber: ++action.payload };
    case ActionType.SET_FRIEND_REQUEST:
      return { ...state, setFriendRequest: action.payload };
    case ActionType.SET_USER_SETTINGS:
      return { ...state, setUserSettings: action.payload };
    case ActionType.TOGGLE_CREATE_GROUP:
      return { ...state, toggleCreateGroup: action.payload };
    case ActionType.SET_MOBILE_NAV:
      return { ...state, setMobileNav: action.payload };
    case ActionType.SET_CHAT_SETTINGS:
      return { ...state, setChatSettings: action.payload };
    case ActionType.SET_IS_MATCH:
      return { ...state, setIsMatch: action.payload };
    case ActionType.SET_MODAL_INVITE:
      return { ...state, setModalInvite: action.payload };
    case ActionType.SET_IS_LOGEDIN:
      return { ...state, setIsLoggingIn: action.payload };
    case ActionType.SIGN_OUT:
      return initialState;
    default:
      return state;
  }
};

export default setReducer;
