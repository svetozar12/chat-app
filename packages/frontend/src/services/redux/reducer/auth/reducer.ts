import { HYDRATE } from 'next-redux-wrapper';
import { IAuth } from './state';
import { ActionType } from '../../types';

export const initialState: IAuth = {
  remember_me: false,
  loginPrompt: false,
};
const authReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default authReducer;
