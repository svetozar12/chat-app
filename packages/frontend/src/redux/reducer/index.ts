import { combineReducers } from "redux";
import reducer from "./authReducer/authReducer";
import setReducer from "./setReducer/setReducer";
import saveInputReducer from "./save_inputReducer/save_inputReducer";
import messageReducer from "./messageReducer/messageReducer";
import { ActionType } from "../types";
const combiReducers = combineReducers({
  authReducer: reducer,
  setReducer,
  messageReducer,
  saveInputReducer,
});

const combReducers = (state: any, action: any) => {
  if (action.type === ActionType.SIGN_OUT) {
    return undefined;
  }
  return combiReducers(state, action);
};

export default combReducers;
