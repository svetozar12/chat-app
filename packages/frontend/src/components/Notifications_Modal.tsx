import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { GrClose } from "react-icons/gr";
import { Socket } from "socket.io-client";
import PendingChats from "./PendingChats";
import { InitialState2 } from "../redux/state";
import { Iinvites } from "../pages/[acc]";
function Notifications({
  contacts,
  socketRef,
  setLocalStatus,
}: {
  contacts: Iinvites[];
  socketRef: Socket | any;
  setLocalStatus: React.Dispatch<React.SetStateAction<string>>;
}) {
  const state = useSelector(
    (state: { setReducer: InitialState2 }) => state.setReducer,
  );

  const dispatch = useDispatch();
  return (
    <div className="fRequests_modal">
      <section
        style={{ position: "relative", textAlign: "center" }}
        className="modal_heading flex"
      >
        <h1 style={{ padding: "0 25%" }}>Notifications</h1>
        <div
          onClick={() => {
            dispatch({
              type: "SET_FRIEND_REQUEST",
              payload: !state.setFriendRequest,
            });
          }}
          style={{ width: "3rem", height: "3rem", margin: "0" }}
          className="circle_border absolute_top_right flex"
        >
          <GrClose style={{ width: "2rem", height: "2rem" }} />
        </div>
      </section>
      <div style={{ overflowY: "scroll" }}>
        {contacts.map((item, index) => {
          return (
            socketRef && (
              <PendingChats
                key={index}
                socketRef={socketRef}
                setLocalStatus={setLocalStatus}
                {...item}
              />
            )
          );
        })}
      </div>
    </div>
  );
}

export default Notifications;
