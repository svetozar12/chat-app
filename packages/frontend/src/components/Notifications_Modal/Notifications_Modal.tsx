import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { GrClose } from "react-icons/gr";
import { Socket } from "socket.io-client";
import PendingChats from "./PendingChats/PendingChats";
import { InitialState2 } from "../../redux/state";
import { Iinvites } from "../../pages/[acc]";
import { css, cx } from "@emotion/css";
function Notifications({
  contacts,
  socketRef,
}: {
  contacts: Iinvites[];
  socketRef: Socket | any;
  setLocalStatus: React.Dispatch<React.SetStateAction<string>>;
}) {
  const state = useSelector((state: { setReducer: InitialState2 }) => state.setReducer);

  const dispatch = useDispatch();
  const checkSize = contacts.filter((element) => element.status != "accepted");

  return (
    <div
      className={css`
        position: fixed;
        flex-direction: column;
        display: flex;
        align-items: center;
        justify-content: !important start;
        z-index: 9999;
        width: 40%;
        background: var(--main-white);
        overflow: hidden;
        left: 0;
        right: 0;
        margin-left: auto;
        margin-right: auto;
        box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
        border-radius: 5px;
        @media (min-width: 1010px) {
          width: 70%;
        }
      `}
    >
      <section
        className={cx(
          "modal_heading flex",
          css`
            position: relative;
            text-align: center;
          `,
        )}
      >
        <h1
          className={css`
            padding: 0 25%;
          `}
        >
          Notifications
        </h1>
        <div
          onClick={() => {
            dispatch({
              type: "SET_FRIEND_REQUEST",
              payload: !state.setFriendRequest,
            });
          }}
          className={cx(
            "circle_border absolute_top_right flex",
            css`
              width: 3rem;
              height: 3rem;
              margin: 0 1rem;
            `,
          )}
        >
          <GrClose
            className={css`
              width: 2rem;
              height: 2rem;
            `}
          />
        </div>
      </section>
      <div
        className={css`
          overflow-y: auto;
          width: 100%;
        `}
      >
        {checkSize.length === 0 && <h1 className="flex">No Chat suggestions</h1>}
        {contacts.map((item, index) => {
          return socketRef && <PendingChats key={index} socketRef={socketRef} {...item} />;
        })}
      </div>
    </div>
  );
}

export default Notifications;
