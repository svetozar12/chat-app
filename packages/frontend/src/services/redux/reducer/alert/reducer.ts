import { AnyAction } from 'redux';
import { IAlert } from 'services/redux/reducer/alert/state';
import { ActionType } from '../../types';

const initialState: IAlert = {
  good: '', // good and bad stand for good alert and bad alert
  bad: '',
};

const alertReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.payload };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.payload };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.payload };
    default:
      return state;
  }
};

export default alertReducer;
