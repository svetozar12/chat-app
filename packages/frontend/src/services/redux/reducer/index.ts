import { AnyAction, combineReducers } from 'redux';
import authReducer from 'services/redux/reducer/auth/reducer';
import { ActionType } from '../types';
import { HYDRATE } from 'next-redux-wrapper';
import wsReducer from 'services/redux/reducer/websocket/reducer';
import messageReducer from 'services/redux/reducer/messages/reducer';

import { IAuth } from 'services/redux/reducer/auth/state';
import { IWebSocket } from 'services/redux/reducer/websocket/state';
import { IMessage } from 'services/redux/reducer/messages/state';
import IInputs from 'services/redux/reducer/inputs/state';
import inputReducer from 'services/redux/reducer/inputs/reducer';
import toggleReducer from 'services/redux/reducer/toggles/reducer';
import { IToggle } from 'services/redux/reducer/toggles/state';
import IInvite from 'services/redux/reducer/invites/state';
import inviteReducer from 'services/redux/reducer/invites/reducer';
import { IAlert } from 'services/redux/reducer/alert/state';
import alertReducer from 'services/redux/reducer/alert/reducer';

export interface STATE {
  auth: IAuth;
  ws: IWebSocket;
  messages: IMessage;
  inputs: IInputs;
  toggle: IToggle;
  invite: IInvite;
  alert: IAlert;
}

const combiReducers = combineReducers({
  auth: authReducer,
  ws: wsReducer,
  messages: messageReducer,
  inputs: inputReducer,
  toggle: toggleReducer,
  invite: inviteReducer,
  alert: alertReducer,
});

const combReducers = (state: any, action: AnyAction) => {
  if (action.type === HYDRATE)
    return {
      ...state,
      ...action.payload,
    };
  if (action.type === ActionType.SIGN_OUT) {
    return undefined;
  }
  return combiReducers(state, action);
};

export default combReducers;
