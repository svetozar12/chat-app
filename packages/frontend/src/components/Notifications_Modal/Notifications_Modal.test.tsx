import Notifications_Modal from "./Notifications_Modal";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { initialState } from "../../redux/reducer/setReducer";
import configureStore from "redux-mock-store";

const socketRef: any = jest.fn();
const mockStore = configureStore([]);
const store = mockStore({
  setReducer: initialState,
});

const setupRender = () => {
  const component = render(
    <Provider store={store}>
      <Notifications_Modal contacts={[{ _id: "31231312", inviter: "ivan", reciever: "greg", status: "declined" }]} socketRef={socketRef} />
    </Provider>,
  );
  return component;
};

describe("Render connected React-redux page", () => {
  beforeEach(() => {
    setupRender();
  });
  it("should create snapshot for <Notifications_Modal/>", () => {
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Notifications_Modal
              contacts={[{ _id: "31231312", inviter: "ivan", reciever: "greg", status: "accepted" }]}
              socketRef={socketRef}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });

  it("should render <Notifications_Modal/>", () => {
    const renderedComponent = screen.getByText("Notifications");
    expect(renderedComponent).toBeInTheDocument();
  });
});
