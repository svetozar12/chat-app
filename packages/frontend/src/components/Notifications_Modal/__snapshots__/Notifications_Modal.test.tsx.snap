// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <Notifications_Modal/> 1`] = `
<div
  className="css-y2iftz"
>
  <section
    className="modal_heading flex css-iteqh8"
  >
    <h1
      className="css-1klbmx9"
    >
      Notifications
    </h1>
    <div
      className="circle_border absolute_top_right flex css-ecfnxd"
      onClick={[Function]}
    >
      <svg
        className="css-vqvu9"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3,3 L21,21 M3,21 L21,3"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </div>
  </section>
  <div
    className="css-1ow1dtq"
  >
    <h1
      className="flex"
    >
      No Chat suggestions
    </h1>
    <div
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
